### PR TITLE
BUG: Fix specviz2d loader with ext keyword

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to address user finding the app crashing when providing `ext` for Specviz2d loader.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-4389)

### CODE REVIEW

File: jdaviz/configs/specviz2d/tests/test_parsers.py [hunk 1]
```diff
@@ -89,6 +91,37 @@ def test_2d_parser_no_unit(specviz2d_helper, mos_spectrum2d):
     assert label_mouseover.icon == 'b'
 
 
+def test_2d_parser_hdulist_ext(tmp_path, specviz2d_helper, mos_spectrum2d_as_hdulist):
```
Comment (pllim) [inline jdaviz/configs/specviz2d/tests/test_parsers.py:94]: This goes through a _similar_ logic route as Cami's problematic case but does not give the exact traceback, probably because I am trying to fake it. But I don't feel like downloading even more data for the test suite and none of the existing remote data tests are triggering Cami's error. 🤷‍♀️

**Comment by camipacifici** [comment]
It loads the requested extension for NIRISS, NIRCam, and NIRSpec (thank you!), but when hovering on the 2D spectrum no information is reported in the top bar and plot options either does not update the 2D spectrum viewer or is extremely slow.

**Comment by pllim** [comment]
Hmm yeah, I guess fixing one case broke another because this parser is so overloaded. Thanks for checking. Turning back into draft.

**Comment by pllim** [comment]
The mouseover is something else.  When you load `jw01895001002_04101_00003_nrcalong_cal.fits` with `ext=8`, its `data.coords` becomes a `PaddedSpectrumWCS`:

```
PaddedSpectrumWCS Transformation

This transformation has 2 pixel and 2 world dimensions

Array shape (Numpy order): None

Pixel Dim  Axis Name  Data size  Bounds
        0  None            None  None
        1  spatial         None  None

World Dim  Axis Name  Physical Type         Units
        0  None       custom:dimensionless  unknown
        1  Offset     None                  unknown

Correlation between pixel and world axes:

           Pixel Dim
World Dim    0    1
        0  yes   no
        1   no  yes
```

And calling `data.coords.pixel_to_world(0, 0)` with that WCS crashes:

`UnitTypeError: SpectralCoord instances require units equivalent to '(Unit("Hz"), Unit("m"), Unit("J"), Unit("1 / m"), Unit("km / s"))', so cannot set it to ''.`

**Comment by camipacifici** [comment]
Now the hovering works and plot options also works fine! When hovering on the 2D spectrum, the indicator does not get propagated to the 1D spectrum, although it gets propagated to the 2D spectrum when hovering on the 1D spectrum.
Everything is in pixels, which probably is expected? Mosviz is (or at least was) able to get the wavelength axis when plotting cal files, but maybe the wavelength was coming from the x1d file which here is not provided?
Thanks!

**Comment by kecnry** [comment]
> Not sure what the expected behavior should be. 

The hover is designed and expected to work in both directions, yes.

https://github.com/spacetelescope/jdaviz/assets/877591/8de424a7-b373-4e59-bd95-459837d263f9

**Comment by pllim** [comment]
To answer your questions from tag-up this morning.

> Does Specviz2d Example Notebook still work with this patch?

@kecnry , yes.

> Why does Mosviz show wavelength for `cal` file?

@camipacifici , that is because for NIRISS, it grabs wavelength from the corresponding WAVELENGTH extension. This seems to be very specific to NIRISS and the logic is buried deep inside Mosviz parser here:

https://github.com/spacetelescope/jdaviz/blob/4423ceb9395380c4ebfd0644b2b9125c48049e2c/jdaviz/configs/mosviz/plugins/parsers.py#L881

https://github.com/spacetelescope/jdaviz/blob/4423ceb9395380c4ebfd0644b2b9125c48049e2c/jdaviz/configs/mosviz/plugins/parsers.py#L889-L901

This logic is inaccessible by Specviz2d.

**Review by camipacifici — APPROVED** [review]
Thank you, looks good to me.

**Review by gibsongreen — APPROVED** [review]
Specviz2d is working now and same with the hover, nice work! (I was curious and had to checkout the test data in the viewer for fun, reminded me of this painting in Chicago Cultural Center)
IMG_0722

**Comment by pllim** [comment]
Thanks for the reviews!
